### PR TITLE
Use "$testapi::password" instead of hardcode in autoyast configuration file

### PR DESCRIPTION
**Pull request by rfan1**
https://progress.opensuse.org/issues/160325

- Verification run: http://openqa.suse.de/tests/14460711#

### CODE REVIEW

File: lib/autoyast.pm [hunk 1]
```diff
@@ -720,6 +721,23 @@ sub expand_variables {
     return $profile;
 }
 
+=head2 adjust_user_password
+
+ adjust_user_password($profile);
+
+ Use default one "$testapi::password" if "PASSWORD" is not defined
```
Comment (jknphy) [inline lib/autoyast.pm:728]: This sentence is not clear, PASSWORD is just a placeholder, not an openQA variable, it is replaced unconditionally wherever it finds that string to replace, "PASSWORD" is never defined. Could you please improve it?

**Review by jknphy — APPROVED** [review]

File: lib/autoyast.pm [hunk 2]
```diff
@@ -720,6 +721,27 @@ sub expand_variables {
...
+ ---
+
+ $profile is the autoyast profile 'autoinst.xml'.
+
+=cut
+
+sub adjust_user_password {
+    my ($profile) = @_;
+    my $user_password = $testapi::password;
+    $profile =~ s/\{\{PASSWORD\}\}/$user_password/g;
```
Comment (Martchus) [inline lib/autoyast.pm:741]: There's probably no use in making this an extra variable:
```suggestion
    $profile =~ s/\{\{PASSWORD\}\}/$testapi::password/g;
```

Comment (rfan1) [reply lib/autoyast.pm]: Fixed

**Review by Martchus — APPROVED** [review]

**Comment by ggardet** [comment]
This PR breaks some tests, such as https://openqa.opensuse.org/tests/4230212#step/console/4 or

**Comment by jknphy** [comment]
Not all the test schedules have prepare_profile.pm

**Comment by ana** [comment]
Breaking Tumbleweed staging tests https://openqa.opensuse.org/tests/4231145#step/console/3

**Pull request merged**